### ISSUE TITLE
Add variant to index

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#22996]
When writing a document to elasticsearch, we should add the variant to both the ES document ID and to the map we put to the index.
We will need to map the variant field as a keyword field in our es-mappings

**Comment by dsilvam** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Falcon-6176e9d907eb0000106ee8e1/issues/dotCMS/core/22996?planning-poker) with Zenhub @freddyucv @victoralfaro-dotcms

**Comment by dsilvam** [comment]
Note to QA:
* Test reindexing
* Check the `_id` field in Kibana has the variant at the end. E.g. `a083e942-5fb0-47ef-9619-6f8f40593474_1_DEFAULT`
* Also in Kibana, check the "variant" field is "keyword". This can be verified by checking the field is "aggregatable"

**Comment by victoralfaro-dotcms** [comment]
Reindexing:
![image](https://user-images.githubusercontent.com/64026161/194893324-33405f12-0604-4cb3-9195-f94b530ffb6b.png)
After reindexing:
![image](https://user-images.githubusercontent.com/64026161/194893451-019fd3a6-5bd2-4e3f-8ba1-26369fa5cc94.png)
Content ID:
![image](https://user-images.githubusercontent.com/64026161/194914143-ec2394d9-a955-4e1e-897e-eee7c5b3e874.png)
Fields:
```
{
          "keywordmapping": {
            "path_match": "categories|tags|conhost|conhostname|wfstep|structurename|contenttype|parentpath|path|urlmap|moduser|owner|metadata.contenttype|metadata.keywords|variant",
            "match_pattern": "regex",
            "mapping": {
              "type": "keyword"
            }
          }
        }
...
"variant": {
  "type": "keyword"
},
"variant_dotraw": {
  "type": "keyword",
  "ignore_above": 8191
}
```

**Comment by bryanboza** [comment]
Fixed, tested on release-22.11 // Docker // Kibana 

![image](https://user-images.githubusercontent.com/2641437/200651790-2cadbf00-96ae-4cac-ab46-3fc965e4c2a9.png)
